src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactService.kt

```diff
@@ -38,4 +40,14 @@ class ContactService(private val contactRepository: ContactRepository, private v
       .also {
         log.debug { "Returning ${it.size} matching Contact records" }
       }
+
+  fun getContactByPrisonNumber(userId: String, prisonNumber: String): Contact =
```

Comment:
Should this be `Contact?` - nullable? It's not an exceptional circumstance when the contact doesn't expect, it's quite ordinary. So modelling it with an exception seems wrong, hence suggetsing the nullable.

Reply:
and if the above is ok - possibly rename the methd to find instead of get?